### DISCUSSION TITLE
I did a midi keyboard for VMIX , but the fastLED light only can use in channel 1

### DISCUSSION THREAD

**Discussion by fareign** [tttapa/Control-Surface#943] [General]
If I use channel 1,the key is ok and  the light also OK.
If I use channel 8, the key is OK,but  the light cant work.
I use midi-ox monitor ,input and out put all is channel 8.

so where is the wrong?

`#include "FastLED.h"
#include "Control_Surface.h"
#define NUM_LEDS 30
constexpr Channel channel = CHANNEL_8;

Array<CRGB, NUM_LEDS> leds{};   
constexpr uint8_t ledpin = 16;  
USBMIDI_Interface usbmidi;
 
NoteRangeFastLED<leds.length> midiled{leds.data, 1 };  
CD74HC4067 mux1{ 6, { 2, 3, 4, 5 } };
CD74HC4067 mux2{ 7, { 8, 9, 10, 14 } };
NoteButton buttons[]{
  //
  { mux2.pin(0), { 1, channel } },
  { mux2.pin(1), { 2, channel } },
  { mux2.pin(2), { 3, channel } },
  { mux2.pin(3), { 4, channel } },
  { mux2.pin(4), { 5, channel } },
  { mux2.pin(5), { 6, channel } },`

**Comment by tttapa** [comment]
You need to pass the channel to the NoteRangeFastLED constructor as well:

```cpp
NoteRangeFastLED<leds.length> midiled{leds.data, { 1, channel } };
```

**Reply by fareign** [reply]
thank you very much!!
